### ISSUE TITLE
Why is the <train_prefix>-class_map.json needed for unsupervised training?

### ISSUE THREAD

**Issue by steffenthoma** [williamleif/GraphSAGE#35]
Hi,

I wanted to use GraphSage for unsupervised training but it seems to need the class membership information (<train_prefix>-class_map.json) although I want to run it in unsupervised mode. Did I overlook something or is this class information always needed?

Thanks in advance!

**Comment by chithangduong** [comment]
My understanding is that the class_map is needed but it is not used for unsupervised training. You can assign random classes to the nodes.

**Comment by williamleif** [comment]
Yes.. this is just an oversight in the way we set things up. It is not really needed, and we hope to remove the unnecessary dependency soon.

**Comment by aisha-deeqa** [comment]
I had the same question. For me, I have two use cases, one is for unsupervised training, and the other for edge classification. I didn't have classes for both.
It would be really helpful if you give this information in the README file.
Thanks!
